store/plugins/preferencesSyncPlugin.ts

```diff
@@ -1,12 +1,91 @@
+import { DisplayPreferencesDto } from '@jellyfin/client-axios';
 import { Plugin } from 'vuex';
-import { AppState } from '..';
+import { AppState } from '~/store';
+import { syncedMutations } from '~/store/clientSettings';
 
 export const preferencesSync: Plugin<AppState> = (store) => {
-  store.subscribe((mutation) => {
-    if (mutation.type.includes('settings/')) {
-      store.dispatch('displayPreferencesApi/updateSettings', null, {
-        root: true
-      });
+  store.subscribe(async (mutation) => {
+    if (
+      syncedMutations.some((syncedMutation) =>
+        mutation.type.includes(syncedMutation)
```

Comment:
I would replace this with an array here that includes the name of the stores we want to have synced. Otherwise, this will need major refactors once we scale the stores further.

With what I mention above about removing the mutation array and switching this logic to the name of the stores we want to have synced, everythig is as simple as:

* Added new settings: **Nothing to do, everything handled automatically**
* Added a new store: **Just add the store name to this plugin**

Otherwise, we need to keep track of two different arrays and I don't think that will scale as good.